iterating-over-data/iterating-over-lists.md

```diff
@@ -0,0 +1,209 @@
+| `for` loop | A kind of loop that will loop over every element in something iterable, such as a list. | for loop, for-loop, for each loop | "I wrote a `for` loop so I can run my function for each element," "I should write a `for` loop here so I can iterate through the list" |
+
+## Syntax for `for` Loops
+
+To iterate over a list, we use the following syntax:
+
+```python
+for my_element in my_list:
+    print(my_element)
+```
+
+| Piece of Code        | Notes                                                                                                                                                                                  |
+| -------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `for`                | `for` is a reserved keyword in Python. Python recognizes `for` as the beginning of a `for` loop.                                                                                       |
+| `my_element`         | **Replace this** part with a name that represents what each element is. This is the name of the local variable we can use in this `for` loop to represent each element, one at a time. |
+| `in`                 | `in` is a reserved keyword in Python. It separates the name of the `my_element` variable and the list we're iterating on.                                                              |
+| `my_list`            | **Replace this** part with the desired list to iterate over. This can be a variable that holds a list, a list literal, or any expression that becomes a list                           |
+| `:`                  | This colon begins the `for` loop's body                                                                                                                                                |
+| ` print(my_element)` | **Replace this** with any code that should execute during each loop. This is the loop's body.                                                                                          |
+
+### About `my_element`
+
+The variable `my_element` gets assigned and re-assigned through a `for` loop.
+
+When a `for` loop starts, `my_element` gets assigned the value of the first element in the list. Then value of `my_element` changes with each following iteration.
+
+1. In the first iteration, the value of `my_element` will be the first element in the list.
+2. In the second iteration, the value of `my_element` will be the second element in the list.
+3. In the third, it will be the third element...
+4. ... etc., until there are no more elements in the list.
+
+We should always change `my_element` to be a variable name appropriate to the context of the list. What does each element in the list represent?
```

Comment:
Taking your suggestion and shortening it

```suggestion
We should always change `my_element` to be a variable name appropriate to the context of the list. Think about what each element in the list represents. If we are iterating over a list of books, we might change `my_element` to `book`. For a list of rooms, the name `room` might be more appropriate.

The name we choose doesn't affect the underlying iteration value in any way, but it can help us remember what data we're processing in the loop body.
```